### PR TITLE
Avoid loading of all assignments objects multiple times when transferring assignments to new user.

### PR DESCRIPTION
**Pull request by shaunakpp**
## What

In `OrganizationEventJob` whenever we transfer assignments from one user to other, in order to check the creator we load all the assignments. This PR is a small optimization to not load these objects and use the `.count` query for the same purpose. It also updates the transfer assignments to first select all assignments of the user, and then update them instead of iterating over all assignments.

### CODE REVIEW

File: app/jobs/organization_event_job.rb [hunk 1]
```diff
@@ -19,20 +19,28 @@ def perform(payload_body)
...
   end
   # rubocop:enable Metrics/AbcSize
 
+  def user_owns_any_assignments?
+    user_owns_assignments? || user_owns_group_assignments?
+  end
+
   def user_owns_assignments?
-    @organization.all_assignments.map(&:creator_id).include? @user.id
+    @organization.assignments.where(creator_id: @user.id).count.positive?
```
Comment (d12) [inline app/jobs/organization_event_job.rb]: Can we do `@organization.assignments.where(creator_id: @user.id).any?` here? This skips counting all the records by applying a `LIMIT 1` to the query.

File: app/jobs/organization_event_job.rb [hunk 2]
```diff
@@ -19,20 +19,28 @@ def perform(payload_body)
...
+  end
+
+  def user_owns_group_assignments?
+    @organization.group_assignments.where(creator_id: @user.id).positive?
```
Comment (d12) [inline app/jobs/organization_event_job.rb]: Similar comment here as my comment above.

File: app/jobs/organization_event_job.rb [hunk 3]
```diff
@@ -19,20 +19,28 @@ def perform(payload_body)
...
   end
 
   def transfer_assignments
```
Comment (d12) [inline app/jobs/organization_event_job.rb]: The stuff in this method was pretty rough, nice refactor. This looks much better!

File: app/jobs/organization_event_job.rb (hunk 1 above)
Comment (d12) [reply app/jobs/organization_event_job.rb]: A quick test in Rails:

```ruby
  irb(main):003:0> User.all.any?
  User Exists (0.4ms)  SELECT  1 AS one FROM "users" LIMIT ?  [["LIMIT", 1]]
```

**Comment by shaunakpp** [comment]
Hi @BenEmdon existing tests are showing full coverage in `SimpleCov`, should I still add tests for the new methods?

**Comment by shaunakpp** [comment]
I realized that there's a slightly larger refactor that involves `OrganizationEventJob` and `OrganizationsController`. So I've pushed it to a new branch and here's the comparison: https://github.com/education/classroom/compare/transfer_assignment_service?expand=1

If y'all are okay with the new changes I'll close this PR and open that.

**Pull request closed without merging**